src/libs/workspace-utils.ts

```diff
@@ -159,10 +159,26 @@ export const isValidWsExportTarget = safeCurry((sourceWs: WorkspaceWrapper, dest
   );
 });
 
-// Returns a message explaining why the user can't edit the workspace, or undefined if they can
-export const editWorkspaceError = ({ accessLevel, workspace: { isLocked } }) => {
-  return cond(
-    [!canWrite(accessLevel), () => 'You do not have permission to modify this workspace.'],
-    [isLocked, () => 'This workspace is locked.']
+export const canEditWorkspace = ({
+  accessLevel,
+  workspace: { isLocked },
+}: {
+  accessLevel: WorkspaceAccessLevel;
+  workspace: { isLocked: boolean };
+}) =>
+  cond<[boolean, string?]>(
+    [!canWrite(accessLevel), () => [false, 'You do not have permission to modify this workspace.']],
+    [isLocked, () => [false, 'This workspace is locked.']],
+    () => [true, undefined]
   );
+
+export const getWorkspaceEditControlProps = ({
+  accessLevel,
+  workspace: { isLocked },
+}: {
+  accessLevel: WorkspaceAccessLevel;
+  workspace: { isLocked: boolean };
+}) => {
+  const [canEdit, disabledMessage] = canEditWorkspace({ accessLevel, workspace: { isLocked } });
+  return canEdit ? {} : { disabled: true, tooltip: disabledMessage };
 };
```

Comment:
Example usage were `canEditWorkspace` to return an Either:
```suggestion
}) => canEditWorkspace({ accessLevel, workspace: { isLocked } })
  .bimap(tooltip => ({ disabled: true, tooltip }), () => ({}))
  .get();
```